src/averageAges.js

```diff
@@ -16,10 +16,19 @@
  */
 function calculateMenAverageAge(people, century) {
   // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => century
+    ? (person.sex === 'm'
+      && Math.ceil(person.died / 100) === century)
+    : person.sex === 'm');
+
+  return getAverageAge(men);
+}
+
+function getAverageAge(peopleSort) {
+  const averageAge = peopleSort.map(person => person.died - person.born)
+    .reduce((sum, x) => sum + x, 0) / peopleSort.length;
+
```

Comment:
You can do it without map method

Reply:
I agree with Myroslav, you can make it more simple
_Hint:_ you may try destructuring inside reduce args

Reply:
I think my code was more readable, but if you want. OK!